service-worker.js

```diff
@@ -91,21 +74,35 @@ function(util) {
     var regexpJS = new RegExp(/\.js/i);
     var regexpCSS = new RegExp(/\.css$/i);
 
-    var regexpContentUrl = new RegExp(/\/(.)\/(.*[^\/]+)$/);
+    var regexpContentUrlWithNamespace = new RegExp(/\/(.)\/(.*[^\/]+)$/);
+    var regexpContentUrlWithoutNamespace = new RegExp(/^([^\/]+)$/);
     var regexpDummyArticle = new RegExp(/dummyArticle\.html$/);
-
-    self.addEventListener('fetch', function(event) {
+    
+    function fetchEventListener(event) {
         console.log('ServiceWorker handling fetch event for : ' + event.request.url);
-        
+
         // TODO handle the dummy article more properly
-        if (regexpContentUrl.test(event.request.url) && !regexpDummyArticle.test(event.request.url)) {
-        
+        if ((regexpContentUrlWithNamespace.test(event.request.url)
+                || regexpContentUrlWithoutNamespace.test(event.request.url))
+            && !regexpDummyArticle.test(event.request.url)) {
+
             console.log('Asking app.js for a content', event.request.url);
             event.respondWith(new Promise(function(resolve, reject) {
-                var regexpResult = regexpContentUrl.exec(event.request.url);
-                var nameSpace = regexpResult[1];
-                var titleName = regexpResult[2];
+                var nameSpace;
+                var titleName;
+                var titleNameWithNameSpace;
                 var contentType;
+                if (regexpContentUrlWithoutNamespace.test(event.request.url)) {
+                    // When the request URL is in the same folder,
```

Comment:
So `event.request.url` could be a relative url? If yes, we have to make it an absolute url relative to a base url, because this will not work correctly if there are subdirectiories in the zim file.